EX_5/src/pepse/world/daynight/Sun.java

```diff
@@ -21,10 +22,15 @@ public static GameObject create(
             float cycleLength){
 
         Renderable sunRenderable = new OvalRenderable(SUN_COLOR);
-        GameObject sunGameObject = new GameObject(Vector2.ZERO, SUN_DIMENSIONS,sunRenderable);
-        sunGameObject.setTag(SUN_TAG);
 
-        sunGameObject.setCoordinateSpace(CoordinateSpace.CAMERA_COORDINATES);
+        GameObject sunGameObject = BackgroundUtils.addBackgroundObject(
+                gameObjects,
+                SUN_DIMENSIONS,
+                Vector2.ZERO,
+                sunRenderable,
+                layer,
+                SUN_TAG);
+
         Vector2 windowCenter = new Vector2(windowDimensions.x()/2, windowDimensions.y()/2);
 
         Transition<Float> locationChanger = new Transition<>(
```

Comment:
the two float type arguments should be saved as constants, the float decleration in the call back is not needed (but doesnt matter that much).